src/Formatter/FRFormatter.php

```diff
@@ -11,13 +11,20 @@
  *
  * Postcodes consist of 5 digits, without separator.
  *
+ * First two digits indicate the department (region)
+ * 01–95 for metropolitan areas
+ * 96–98 for overseas territories
+ *
  * @see https://en.wikipedia.org/wiki/List_of_postal_codes
+ * @see https://en.wikipedia.org/wiki/Postal_codes_in_France
  */
 class FRFormatter implements CountryPostcodeFormatter
 {
+
     public function format(string $postcode) : ?string
     {
-        if (preg_match('/^[0-9]{5}$/', $postcode) !== 1) {
+
+        if (preg_match('/^(?:0[1-9]|[1-8]\d|9[0-8])\d{3}$/', $postcode) !== 1) {
```

Comment:
Please avoid mixing `\d` with ranges, I think it hurts readability. Let's keep `[0-9]` everywhere.

Reply:
I agree with this perspective, as even if the given numbers are not in a valid range, using [0-9] enhances clarity.